openapi_python_client/parser/openapi.py

```diff
@@ -258,37 +279,21 @@ def _add_parameters(
             if isinstance(prop, ParseError):
                 return ParseError(detail=f"cannot parse parameter of endpoint {endpoint.name}", data=prop.data), schemas
 
-            if prop.python_name in used_python_names:
-                duplicate, duplicate_location = used_python_names[prop.python_name]
-                if duplicate.python_name == prop.python_name:  # Existing should be converted too for consistency
-                    duplicate.set_python_name(f"{duplicate.python_name}_{duplicate_location}", config=config)
-                prop.set_python_name(f"{prop.python_name}_{param.param_in}", config=config)
-            else:
-                used_python_names[prop.python_name] = (prop, param.param_in)
+            for location, parameters_dict in parameters_by_location.items():
+                existing_prop: Optional[Property] = parameters_dict.get(prop.name)
+                same_location: bool = location == param.param_in
 
-            endpoint.relative_imports.update(prop.get_imports(prefix="..."))
+                if existing_prop and not same_location:
+                    # Existing should be converted too for consistency
+                    endpoint.used_python_identifiers.remove(existing_prop.python_name)
+                    existing_prop.set_python_name(new_name=f"{existing_prop.name}_{location}", config=config)
+                    endpoint.used_python_identifiers.add(existing_prop.python_name)
 
-            if param.param_in == oai.ParameterLocation.QUERY:
-                endpoint.query_parameters.append(prop)
-            elif param.param_in == oai.ParameterLocation.PATH:
-                endpoint.path_parameters.append(prop)
-            elif param.param_in == oai.ParameterLocation.HEADER:
-                endpoint.header_parameters.append(prop)
-            elif param.param_in == oai.ParameterLocation.COOKIE:
-                endpoint.cookie_parameters.append(prop)
-            else:
-                return ParseError(data=param, detail="Parameter must be declared in path or query"), schemas
-
-        name_check = set()
-        for prop in itertools.chain(
-            endpoint.query_parameters, endpoint.path_parameters, endpoint.header_parameters, endpoint.cookie_parameters
-        ):
-            if prop.python_name in name_check:
-                return (
-                    ParseError(data=data, detail=f"Could not reconcile duplicate parameters named {prop.python_name}"),
-                    schemas,
-                )
-            name_check.add(prop.python_name)
+                    prop.set_python_name(new_name=f"{param.name}_{param.param_in}", config=config)
+
+            endpoint.relative_imports.update(prop.get_imports(prefix="..."))
+            parameters_by_location[param.param_in].setdefault(prop.name, prop)
```

Comment:
I want to use `dict.setdefault()` here because first the `Endpoint` is created to which `Operation` parameters are added. Then added `PathItem` parameters to the `EndpointCollection`. `Operation` parameters has priority over `PathItem` parameters. `dict.setdefault()` allows not to override `Operation` parameters if they are added.

Reply:
I refactored it. And finally used this 😅